src/gridcoin/scraper/scraper_net.cpp

```diff
@@ -685,42 +687,44 @@ bool CScraperManifest::RecvManifest(CNode* pfrom, CDataStream& vRecv)
 
     CScraperManifest_shared_ptr manifest = std::shared_ptr<CScraperManifest>(new CScraperManifest());
 
-    LOCK(cs_mapManifest);
+    LOCK2(cs_mapManifest, cs_mapParts);
 
     const auto it = mapManifest.emplace(hash, manifest);
 
-    LOCK2(cs_mapParts, manifest->cs_manifest);
-
-    // The phash in the manifest points to the actual hash which is the index to the element in the map.
-    manifest->phash = &it.first->first;
-
-    try
     {
-        manifest->UnserializeCheck(vRecv, banscore);
-    } catch (bool& e)
-    {
-        mapManifest.erase(hash);
-        LogPrint(BCLog::LogFlags::MANIFEST, "invalid manifest %s received", hash.GetHex());
+        LOCK(manifest->cs_manifest);
 
-        if (pfrom)
+        // The phash in the manifest points to the actual hash which is the index to the element in the map.
+        manifest->phash = &it.first->first;
+
+        try
         {
-            LogPrintf("WARNING: CScraperManifest::RecvManifest: Invalid manifest %s received from %s. Increasing banscore "
-                      "by %u.", hash.GetHex(), pfrom->addr.ToString(), banscore);
-            pfrom->Misbehaving(banscore);
-        }
-        return false;
-    } catch(std::ios_base::failure& e)
-    {
-        mapManifest.erase(hash);
-        LogPrint(BCLog::LogFlags::MANIFEST, "invalid manifest %s received", hash.GetHex());
+            manifest->UnserializeCheck(vRecv, banscore);
+        } catch (bool& e)
```

Comment:
Why are we catching a bool here?

Reply:
🤷‍♂️ . This part was written by Tomas a long time ago. Even though I have heavily edited sections of this code since. We can just have the method return a bool directly....

Reply:
I changed it to a bool and also put a nodiscard qualifier around it, since the status of the function MUST be used now. Note that it still must be used inside a try/catch, because the basic serialization could fail due to a malformed incoming message. I think that was @tomasbrod's original rationale around throwing the boolean rather than returning it... You have to run the thing inside a try/catch anyway...

So this is a close call on style quite frankly.

Reply:
Really no reason why false should be returned or thrown. It is unusual to throw false and there is tiny stack unwinding penalty, that's true. But this was easy solution at the time of writing. Declaring exception class was not straightforward and returning false was risking missing the error condition during testing. Plus it was consistent with other errors - namely truncated message, that throw, instead of returning, as James said. Guess I should have used std::something exception, that did not occur to me, nor reviewers :)

Reply:
Tomas the code works great in the original form or this way too. This is really a difference of style IMHO… :)